auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -289,30 +248,22 @@ private Uri buildAuthorizeUri() {
     }
 
     private void addPKCEParameters(Map<String, String> parameters, String redirectUri) {
-        if (!shouldUsePKCE()) {
-            return;
-        }
         try {
             createPKCE(redirectUri);
             String codeChallenge = pkce.getCodeChallenge();
             parameters.put(KEY_CODE_CHALLENGE, codeChallenge);
             parameters.put(KEY_CODE_CHALLENGE_METHOD, METHOD_SHA_256);
             Log.v(TAG, "Using PKCE authentication flow");
         } catch (IllegalStateException e) {
```

Comment:
what conditions are required for this exception to be thrown?

Reply:
When creating an instance of `PKCE`, it generates the code challenge, which wraps and rethrows if it cannot obtain the signature:

```java
try {
        MessageDigest md = MessageDigest.getInstance(SHA_256);
        md.update(input, 0, input.length);
        signature = md.digest();
} catch (NoSuchAlgorithmException e) {
        Log.e(TAG, "Failed to get SHA-256 signature", e);
        throw new IllegalStateException("Failed to get SHA-256 signature", e);
}
```

Previously this logged an error that PKCE isn't available, and instead will fall back to the token response_type. Related to your other questions about how to best handle something that should never happen, and cannot be recovered from, we could just ignore this exception and let it propagate if you prefer.